### PR TITLE
fix for 'page level bookmarking' bug, convert to ES6 class

### LINKED ISSUE #2659: adapt-contrib-bookmarking: location not restored if using page level locations
### Subject of the issue/enhancement/features
Due to the changes made to the `className` in 62d75442ef7000bc800999b93b533bb83c6068ed, if bookmarking is enabled to `"_level": "page"`, the lesson location is not restored. The id to restore is found on the menu item, so the extension already thinks it is on screen and returns out of the `restoreLocation` method.

### PR DESCRIPTION
**Pull request by moloko**
fixes https://github.com/adaptlearning/adapt_framework/issues/2659 and https://github.com/adaptlearning/adapt_framework/issues/3036

### CODE REVIEW

**Comment by moloko** [comment]
right, this should be good to go now... tested in Firefox, Chrome and IE11. 'Page level bookmarking' bug now fixed and the 'don't show prompt if element already on screen' functionality now working as it should.

File: js/adapt-contrib-bookmarking.js [hunk 1]
```diff
@@ -1,236 +1,234 @@
...
+  onAdaptInitialize() {
+    if (!this.checkIsEnabled()) return;
+    this.setupEventListeners();
+    this.checkRestoreLocation();
+  }
+
+  checkIsEnabled() {
+    const courseBookmarkModel = Adapt.course.get('_bookmarking');
+    if (!courseBookmarkModel || !courseBookmarkModel._isEnabled) return false;
+    if (!Adapt.offlineStorage) return false;
```
Comment (eleanor-heath) [inline js/adapt-contrib-bookmarking.js]: this condition `!Adapt.offlineStorage` could go into the if statement on line 22

File: js/adapt-contrib-bookmarking.js [hunk 2]
```diff
@@ -1,236 +1,234 @@
...
+          promptText: buttons.no || 'No',
+          _callbackEvent: 'bookmarking:cancel'
+        }
+      ]
+    });
+  }
+
+  navigateToPrevious() {
+    _.defer(async () => {
+      var isSinglePage = (Adapt.contentObjects.models.length === 1);
```
Comment (eleanor-heath) [inline js/adapt-contrib-bookmarking.js]: const instead of var

File: js/adapt-contrib-bookmarking.js [hunk 3]
```diff
@@ -1,236 +1,234 @@
...
-        if (!isViewAPageChild ) continue;
+  captureViews(view) {
+    this.watchViews.push(view);
+  }
 
-        var element = $('.' + view.model.get('_id'));
-        var measurements = element.onscreen();
+  setLocationID(id) {
+    if (!Adapt.offlineStorage) return;
+    if (this.currentLocationID === id) return;
```
Comment (eleanor-heath) [inline js/adapt-contrib-bookmarking.js:193]: could maybe combine these 2 if statements

**Review by eleanor-heath — APPROVED** [review]

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

File: js/adapt-contrib-bookmarking.js (hunk 1 above)
Comment (moloko) [reply js/adapt-contrib-bookmarking.js]: Now that you mention it I'm actually struggling to think of any condition under which `Adapt.offlineStorage` wouldn't exist... might take that check out altogether.

**Review by eleanor-heath — APPROVED** [review]

**Review by oliverfoster — APPROVED** [review]

**Pull request merged**